### ISSUE TITLE
UTM Params not being passed when signing up with Google from landing page(s)

### ISSUE THREAD

**Issue by tythorsen** [closeio/webflow-js#6]
**TL;DR(W):** We have a marketing site landing page with a Sign up with Google button for which we're failing to pass our UTM params. We need to add some code to the marketing site JS to ensure they are passed correctly.

### Background

@jsnyder18 and @mjtay10r in marketing (with an assist from @morrismatts) have identified a gap in our conversion tracking. You can view Josh's original [Loom video](https://www.loom.com/share/3ae13074e56b435e87fb268549e2f159?sid=49098f83-983e-40e5-9259-3bc5c0142928) to get some additional context.

Basically, we have one landing page ([www.close.com/get-started](https://www.close.com/get-started)) that allows visitors to sign up with Google in addition to a normal sign-up form that pre-populates the in-app signup form. 

<img width="1362" alt="Image" src="https://github.com/user-attachments/assets/ba48eed3-7f00-42b7-90fd-864d2b2c5f35" />

Signups that use the form work fine. Signups that use the Google OAuth option are missing UTMs. This is due to the fact that the button on this page is linking directly to our Google signup url [https://app.close.com/auth/google/sign-up/?app_platform=web](https://app.close.com/auth/google/sign-up/?app_platform=web) and we expect UTMs to be appended to the URL in this scenario.

### Proposed Solution

When users signup with Google in the app, we have some logic in [js/components/guest/utils.ts](https://github.com/closeio/close-ui/blob/80111f7a0acfb8f29b792d0c4dccab11b6d52a1b/js/components/guest/utils.ts) that appends a few things like [analytics context](https://github.com/closeio/close-ui/blob/80111f7a0acfb8f29b792d0c4dccab11b6d52a1b/js/components/guest/utils.ts#L50) and [UTM values](https://github.com/closeio/close-ui/blob/80111f7a0acfb8f29b792d0c4dccab11b6d52a1b/js/components/guest/utils.ts#L42) as search params before redirecting to Google. We need to port a simplified version of this code into Webflow for the marketing site.
* The app has a few abstractions to handle different types of providers and OAuth types. We can directly address the Google signup case.
* We should probably write our solution using a global click handler that targets any link to the Google OAuth URL so that marketing can easily add this functionality to other landing pages in the future.

**Comment by jsnyder18** [comment]
Hi @tythorsen and @bartgryszko - hope you both had a good holiday. Wanted to followup on this ticket - any potential update? Thanks so much!

cc: @mjtay10r
